pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -2568,8 +2568,39 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
                                                     wires, adj);
     }
 
+    template <class PrecisionT>
+    [[nodiscard]] static auto
+    applyNCGeneratorPSWAP(std::complex<PrecisionT> *arr, std::size_t num_qubits,
+                          const std::vector<std::size_t> &controlled_wires,
+                          const std::vector<bool> &controlled_values,
+                          const std::vector<std::size_t> &wires,
+                          [[maybe_unused]] const bool adj) -> PrecisionT {
+        constexpr std::complex<PrecisionT> zero{0.0};
```

Comment:
```suggestion
        using ComplexT = std::complex<PrecisionT> 
        constexpr ComplexT zero{0.0};
```
Could you change the type in the rest of the function, please?

Reply:
hahaha are only two definition :sweat_smile: